tests/testthat/test_preprocess_ww_data.R

```diff
@@ -24,6 +25,87 @@ test_that("Function returns dataframe with correct columns", {
   checkmate::expect_names(names(processed), must.include = expected_cols)
 })
 
+# Test that can pass either integer or character site names
+ww_data_char <- tibble::tibble(
+  date = lubridate::ymd(rep(c("2023-11-01", "2023-11-02"), 2)),
+  site = c("1", "1", "2", "2"),
+  lab = c(1, 1, 1, 1),
+  conc = log(c(345.2, 784.1, 401.5, 681.8)),
+  lod = log(c(20, 20, 15, 15)),
+  site_pop = c(rep(1e6, 2), rep(3e5, 2)),
+  location = c(rep("MA", 4))
+)
+
+ww_data_int <- tibble::tibble(
+  date = lubridate::ymd(rep(c("2023-11-01", "2023-11-02"), 2)),
+  site = c(1, 1, 2, 2),
+  lab = c(1, 1, 1, 1),
+  conc = log(c(345.2, 784.1, 401.5, 681.8)),
+  lod = log(c(20, 20, 15, 15)),
+  site_pop = c(rep(1e6, 2), rep(3e5, 2)),
+  location = c(rep("MA", 4))
+)
+
+ww_data_int_alt <- tibble::tibble(
+  date = lubridate::ymd(rep(c("2023-11-01", "2023-11-02"), 2)),
+  site = c(5, 5, 1, 1),
+  lab = c(1, 1, 1, 1),
+  conc = log(c(345.2, 784.1, 401.5, 681.8)),
+  lod = log(c(20, 20, 15, 15)),
+  site_pop = c(rep(1e6, 2), rep(3e5, 2)),
+  location = c(rep("MA", 4))
+)
+
+test_that("Function returns dataframe with correct site indices", {
+  processed_int <- preprocess_ww_data(ww_data_int,
+    conc_col_name = "conc",
+    lod_col_name = "lod"
+  )
+  processed_char <- preprocess_ww_data(ww_data_char,
+    conc_col_name = "conc",
+    lod_col_name = "lod"
+  )
+  processed_int_alt <- preprocess_ww_data(ww_data_int_alt,
+    conc_col_name = "conc",
+    lod_col_name = "lod"
+  )
+  # site indices should be the same even if sites are not ordered or are
+  # characters
+  expect_equal(processed_int$site_index, processed_char$site_index)
+  expect_equal(processed_int_alt$site_index, processed_int$site_index)
+  expect_equal(processed_int_alt$site_index, processed_char$site_index)
+  # Ordering shouldn't change even if site integers not in order
+  expect_equal(processed_int_alt$site, ww_data_int_alt$site)
+})
+
+ww_data_w_repeats <- tibble::tibble(
+  date = lubridate::ymd(
+    rep(c("2023-11-01", "2023-11-02"), 2),
+    "2023-11-02"
+  ),
+  site = c("1", "1", "2", "2", "2"),
+  lab = c(1, 1, 1, 1, 1),
+  conc = log(c(345.2, 784.1, 401.5, 681.8, 681.8)),
+  lod = log(c(20, 20, 15, 15, 15)),
+  site_pop = c(rep(1e6, 2), rep(3e5, 3)),
+  location = c(rep("MA", 5))
+)
+
+test_that("Function returns an error if there are repeated values", {
+  expect_error(
```

Comment:
check that the error message is the one you expect?

Reply:
I couldn't figure out how to do this for a really long error message like:
```
"! The data has more than one observation per `lab-site-day`
Package expects either at most one
wastewater observation per a given a site, lab,
and sample collection date. Got date(s) with
more than one observation for a given site and lab.
ℹ Multiple observations in a `lab-site-day` are not
currently supported." 
```

Reply:
Should be able to check for a partial match via `regexp`, but even better if the expected error is one of the ones for which you wrote a custom class, you can and should check for that. See https://testthat.r-lib.org/reference/expect_error.html#testing-message-vs-class

Reply:
Whoops yeah didn't realize you could just do part of it which I think is sufficient here!